src/Sarif.FunctionalTests/ConverterTestData/FxCop/FxCopReportAllMessageStates.xml.sarif

```diff
@@ -0,0 +1,357 @@
+{
+  "version": "1.0.0-beta.4",
+  "runs": [
+    {
+      "tool": {
+        "name": "FxCop"
+      },
+      "files": {
+        "file:///C:/Users/user/Documents/Visual Studio 2015/Projects/WpfApplication5/WpfApplication5/bin/Debug/WpfApplication5.exe": [
+          {
+            "mimeType": "application/octet-stream"
+          }
+        ],
+        "file:///c:/users/user/documents/visual studio 2015/Projects/WpfApplication5/WpfApplication5/Foo.cs": [
+          {
+            "mimeType": "text/x-csharp"
+          }
+        ]
+      },
+      "logicalLocations": {
+        "WpfApplication5": [
+          {
+            "name": "WpfApplication5",
+            "kind": "namespace"
+          }
+        ],
+        "wpfapplication5.exe": [
+          {
+            "name": "wpfapplication5.exe",
+            "kind": "module"
+          }
+        ],
+        "WpfApplication5.Foo": [
+          {
+            "name": "wpfapplication5.exe",
+            "kind": "module"
+          },
+          {
+            "name": "WpfApplication5",
+            "kind": "namespace"
+          },
+          {
+            "name": "Foo",
+            "kind": "type"
+          }
+        ],
+        "WpfApplication5.Foo.Baz()": [
+          {
+            "name": "wpfapplication5.exe",
+            "kind": "module"
+          },
+          {
+            "name": "WpfApplication5",
+            "kind": "namespace"
+          },
+          {
+            "name": "Foo",
+            "kind": "type"
+          },
+          {
+            "name": "#Baz()",
+            "kind": "member"
+          }
+        ],
+        "WpfApplication5.MainWindow.System.Windows.Markup.IComponentConnector.Connect(System.Int32,System.Object)": [
+          {
+            "name": "wpfapplication5.exe",
+            "kind": "module"
+          },
+          {
+            "name": "WpfApplication5",
+            "kind": "namespace"
+          },
+          {
+            "name": "MainWindow",
+            "kind": "type"
+          },
+          {
+            "name": "#System.Windows.Markup.IComponentConnector.Connect(System.Int32,System.Object)",
+            "kind": "member"
+          }
+        ],
+        "WpfApplication5.Properties.Resources": [
+          {
+            "name": "wpfapplication5.exe",
+            "kind": "module"
+          },
+          {
+            "name": "WpfApplication5.Properties",
+            "kind": "namespace"
+          },
+          {
+            "name": "Resources",
+            "kind": "type"
+          }
+        ],
+        "WpfApplication5.Properties.Resources..ctor()": [
+          {
+            "name": "wpfapplication5.exe",
+            "kind": "module"
+          },
+          {
+            "name": "WpfApplication5.Properties",
```

Comment:
Should this be two namespaces?

```json
{
  "name": "WpfApplication5",
  "kind": "namespace"
},
{
  "name": "Properties",
  "kind": "namespace"
}
```

If we specify that namespace components should be separated, then we can can ask the question "How many issues are in namespace N _and all of its sub-namespaces_", without having to parse a compound name. If we specify that namespace components should be combined (as you did here), then we can ask the question "How many issues are in namespace N, _excluding its sub-namespaces_", without having to examine the next logical location component to ensure that it isn't a namespace.

So the questions are:

1. Which is the more useful question to be able to ask easily?
2. If there's no clear answer to (1), do we give producers the choice of whether to separate or combine the components?